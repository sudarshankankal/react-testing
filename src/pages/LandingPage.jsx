import React from "react";
import { Col, Container, Row } from "react-bootstrap";
import FetchData from "../components/FetchData";
import { Login } from "../components/Login";
import { SignUp } from "../components/SignUp";

function LandingPage() {
  return (
    <div className="box">
      <Container fluid>
        <Row>
          <Col>
            <Login />
          </Col>
          <Col>
            <SignUp />
          </Col>
          <Col><FetchData loader={false}/></Col>
        </Row>
      </Container>
    </div>
  );
}

export default LandingPage;
