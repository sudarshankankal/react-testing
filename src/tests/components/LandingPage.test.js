import React from "react";
import { shallow } from "enzyme";
import LandingPage from "../../pages/LandingPage";

describe("Given LandingPage", () => {
  
  test("Check Snapshot for LandingPage", () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });

  test("class box should render properly",() => {
      const wrapper = shallow(<LandingPage />);
      expect(wrapper.hasClass('box')).toBe(true);
  })

  test("it should render SignIn/SignUp/List", () => {
    const wrapper = shallow(<LandingPage />);
    const Login = wrapper.find("Login").exists();
    const SignUp = wrapper.find("SignUp").exists();
    const list = wrapper.find("FetchData").exists();
    expect(Login).toBe(true);
    expect(SignUp).toBe(true);
    expect(list).toBe(true);
  })
});


