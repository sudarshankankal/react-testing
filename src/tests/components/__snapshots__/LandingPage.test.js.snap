// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given LandingPage Check Snapshot for LandingPage 1`] = `
<div
  className="box"
>
  <Container
    fluid={true}
  >
    <Row
      noGutters={false}
    >
      <Col>
        <Login />
      </Col>
      <Col>
        <SignUp />
      </Col>
      <Col>
        <FetchData
          loader={false}
        />
      </Col>
    </Row>
  </Container>
</div>
`;
